paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -178,6 +182,12 @@ internal class PaymentOptionsViewModel @Inject constructor(
         clearErrorMessages()
 
         selection.value?.let { paymentSelection ->
+            if (paymentSelection is PaymentSelection.Link) {
+                val linkState = args.state.paymentMethodMetadata.linkState
+                val prominenceEnabled = linkState !== null && linkProminenceFeatureProvider
+                    .showVerificationOnFlowControllerLinkSelection(linkState)
+                logger.debug("Link prominence enabled: $prominenceEnabled")
+            }
```

Comment:
follow-up work: show the dialog if `showVerificationOnFlowControllerLinkSelection` returns true, keep the existing behavior otherwise.